src/EventFactory.php

```diff
@@ -0,0 +1,141 @@
+use Zend\Diactoros\ServerRequestFactory;
+
+final class EventFactory
+{
+    /**
+     * @var SerializerInterface The serializer
+     */
+    private $serializer;
+
+    /**
+     * @var RepresentationSerializerInterface The representation serializer
+     */
+    private $representationSerializer;
+
+    /**
+     * @var Options The Sentry options
+     */
+    private $options;
+
+    /**
+     * @var string The Sentry SDK identifier
+     */
+    private $sdkIdentifier;
+
+    /**
+     * EventFactory constructor.
+     *
+     * @param SerializerInterface               $serializer               The serializer
+     * @param RepresentationSerializerInterface $representationSerializer The serializer for function arguments
+     * @param Options                           $options                  The SDK configuration options
+     * @param string                            $sdkIdentifier            The Sentry SDK identifier
+     */
+    public function __construct(SerializerInterface $serializer, RepresentationSerializerInterface $representationSerializer, Options $options, string $sdkIdentifier)
+    {
+        $this->serializer = $serializer;
+        $this->representationSerializer = $representationSerializer;
+        $this->options = $options;
+        $this->sdkIdentifier = $sdkIdentifier;
+    }
+
+    /**
+     * @param array $payload The data to be attached to the Event
+     *
+     * @return Event
+     */
+    public function create(array $payload): Event
+    {
+        try {
+            $event = new Event($this->sdkIdentifier);
+        } catch (\Throwable $error) {
+            throw new \RuntimeException('Unable to instantiate an event', 0, $error);
+        }
+
+        $event->setServerName($this->options->getServerName());
+        $event->setRelease($this->options->getRelease());
+        $event->getTagsContext()->merge($this->options->getTags());
+        $event->setEnvironment($this->options->getEnvironment());
+
+        if (isset($payload['transaction'])) {
+            $event->setTransaction($payload['transaction']);
+        } else {
+            $request = ServerRequestFactory::fromGlobals();
+            $serverParams = $request->getServerParams();
+
+            if (isset($serverParams['PATH_INFO'])) {
+                $event->setTransaction($serverParams['PATH_INFO']);
+            }
+        }
+
+        if (isset($payload['logger'])) {
+            $event->setLogger($payload['logger']);
+        }
+
+        $message = $payload['message'] ?? null;
+        $messageParams = $payload['message_params'] ?? [];
+
+        if (null !== $message) {
+            $event->setMessage(substr($message, 0, Client::MESSAGE_MAX_LENGTH_LIMIT), $messageParams);
+        }
+
+        if (isset($payload['exception']) && $payload['exception'] instanceof \Throwable) {
+            $this->addThrowableToEvent($event, $payload['exception']);
+        } elseif ($this->options->shouldAttachStacktrace()) {
```

Comment:
We have to be careful here, the docs say that this really only works for `captureMessage`. In theory if someone calls `captureEvent` this will also trigger.
So maybe this has to go back into `captureMessage`.

Reply:
That's not an issue here. There is only `captureMessage` and `captureException`, and the latter sets the `exception` key, so it will trigger the first `if` branch.

`captureLastError` goes through `captureException` too, and `captureEvent` would behave like `captureMessage`.

Reply:
>`captureEvent` would behave like `captureMessage`

If I don't recall wrong it means that `captureMessage` will use the `shouldAttachStacktrace` option which is wrong

Reply:
Moving that piece of code back would mean loosing the refactoring in term of deps splitting. I'll think about it a bit more...